### PR TITLE
New poke scan modes

### PR DESCRIPTION
**Pull request by madBeavis**
- Create pokemon mode where it hops to newly spawned monster, called jumpy (for now), used for better timers on spawns.  Give it a geofence and assign enough devices, it should give one 28+ minimum spawn timers.
- Create pokemon mode for use to narrow down and find tth, called findy.  
- No calculation of routes necessary, it does it all automatically when provided with a geofence.

## Description

Created new pokemon scanning modes

## Motivation and Context

Rounds don't produce long enough timers for what I require.  Since I am not a spoofer, I want 28+ minute timers.  Jumpy pokemon solves this, assuming you tune the setup and have enough devices.  If you are device poor, this mode is not for you and you would be better with circle pokemon.

Does address or attempt to fix any open issue.

## How Has This Been Tested?

My own personnel instances for several months now.  I did my hometown instance a few months ago when I swapped to Atlas.  I am working on finding tth for both my instances as I type, due to spawn changes.

On my bare metal server at home.  I run, observe, debug, modify and repeat.  I refuse to use docker, so all dev and testing done from command line with swift run and pm2.  I don't develop on Mac environment, so unable to run lint.

I do not think it affects other areas of the code as it creates whole new methods of scanning for pokemon

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,98 +0,0 @@
-version: '3.1'
```
Comment (Fabio1988) [inline docker-compose.yml:1]: you don't need to delete this file if you don't use docker for testing :rofl:

**Comment by kamieniarz** [comment]
Just a hint for anyone who never used MAD that actually has something like called `prioQ` which is pretty much the same as `jumpy` and would like to improve or reconsider things ;) It basically creates a queue based of events (eggs hatches, spawns etc)

So basically `prioQ` is an addition to mons scanning type of scans and it's used to generate priority routes based on despawn times. It has parameters like:
- `delay_after_prio_event` - scan after x seconds of an event, eg spawnpoint says mon should spawn at x:49, we set this param to 60s so it will be added to queue at x:50
- `max_clustering` - max number of events in same cluster, eg. max 3 spawnpoints 
- `priority_queue_clustering_timedelta` - timedelta for clustering
- `remove_from_queue_backlog` - remove old events from queue after x time, eg. when instance is not catching up (something we want to avoid for sure)

**Comment by madBeavis** [comment]
intention was not to copy mad, RMD/Atlas operates at high rate of speed so i did not add things like clustering.  Jumpy offers an option for folks that require better timers than circle scan modes offered.  one will get less spawns than circle, but will get better timers.

tunable option are on the list.  but with power to change values comes the responsibility to know ramifications of such changes.  i also wanted to just get things out for people to use with new spawn points.

i did write up some more info on my wiki.  i intend to get some of the more basic information into the rdm wiki.

File: Sources/RealDeviceMapLib/Controller/DBController.swift [hunk 2]
```diff
@@ -336,7 +336,7 @@ public class DBController {
 
             // Schema
             //  swiftlint:disable:next line_length
-            let commandSchema = Shell("bash", "-c", mysqldumpCommand + " --column-statistics=0 --set-gtid-purged=OFF --skip-triggers --add-drop-table --skip-routines --no-data \(self.database) \(tablesSchema) -h \(self.host) -P \(self.port) -u \(self.rootUsername) -p\(self.rootPassword.stringByReplacing(string: "\"", withString: "\\\"")) > \(backupFileSchema.path)")
```
Comment (Fabio1988) [inline Sources/RealDeviceMapLib/Controller/DBController.swift:339]: why did you changed backup scripts?!
are you running mysql 5.7 or 8?

File: docker-compose.yml (hunk 1 above)
Comment (Fabio1988) [reply docker-compose.yml:1]: pls add file again

File: Sources/RealDeviceMapLib/Controller/DBController.swift (hunk 2 above)
Comment (madBeavis) [reply Sources/RealDeviceMapLib/Controller/DBController.swift:339]: had to modify, mariadb 10.6.10 didn't like it at all.

Comment (Fabio1988) [reply Sources/RealDeviceMapLib/Controller/DBController.swift:339]: so we should fix it for both.. because when you remove it it will crash on mysql ;)

**Comment by Fabio1988** [comment]
fixed linting and rebased on development

**Comment by Fabio1988** [comment]
Findy > Auto TTH
Jumpy > Auto Spawnpoint

File: Sources/RealDeviceMapLib/Controller/DBController.swift (hunk 2 above)
Comment (madBeavis) [reply Sources/RealDeviceMapLib/Controller/DBController.swift:339]: finally back to work on things.  i don't know how to make it compatible.

Comment (madBeavis) [reply Sources/RealDeviceMapLib/Controller/DBController.swift:339]: did find this in perfect mysql, need to see what data it gives back, if anything like the version from shell one should be able to check and take action based on such.

public static func clientInfo() -> String

File: Sources/RealDeviceMapLib/Controller/InstanceControllers/AutoInstanceController.swift [hunk 3]
```diff
@@ -271,13 +309,108 @@ class AutoInstanceController: InstanceControllerProto {
...
+            return task
+        case .tth:
+            // get route like for findy, specify fence and use tth = null
+            // with each gettask, just increment to next point in list
+            // requery the route every ???? min, set with cache above
+            // run until data length == 0, then output a message to tell user done
+            // since we actually care about laptime, use that variable
+            findyLock.lock()
+
+            let hit = findyCache.get(id: self.name) ?? 0
```
Comment (Fabio1988) [inline Sources/RealDeviceMapLib/Controller/InstanceControllers/AutoInstanceController.swift:367]: what's that cache used for? it contains one element with value 1
You are only using it for being sure that init for findy coords is only run once at starting instance?!

Comment (madBeavis) [reply Sources/RealDeviceMapLib/Controller/InstanceControllers/AutoInstanceController.swift]: cache is used in findy for requery of data.  if the init wasn't perfect as workers move around the area, they find more spawnpoints.  i set the time to requery in initialization of the cache, 5min.  some had said less would be better, but i think it should be a variable one can config.

**Review by Fabio1988 — APPROVED** [review]

**Pull request merged**